### PR TITLE
feat: derive more traits

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

File: crates/revm/src/evm_impl.rs [hunk 1]
```diff
@@ -31,6 +33,17 @@ pub struct EVMImpl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> {
...
 }
 
+impl<GSPEC, DB, const INSPECT: bool> fmt::Debug for EVMImpl<'_, GSPEC, DB, INSPECT>
+where
+    GSPEC: Spec,
+    DB: Database + fmt::Debug,
+    DB::Error: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("EVMImpl").field("data", &self.data).finish()
```
Comment (mattsse) [inline crates/revm/src/evm_impl.rs:43]:
```suggestion
        f.debug_struct("EVMImpl").field("data", &self.data).finish_non_exhaustive()
```

**Review by mattsse — APPROVED** [review]
lgtm,

some derives like hash may not be needed but can't hurt

File: crates/interpreter/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,7 @@
+#![warn(missing_debug_implementations, unreachable_pub)]
+#![deny(unused_must_use, rust_2018_idioms)]
 #![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
```
Comment (rakita) [inline crates/interpreter/src/lib.rs:4]: nice!

File: crates/revm/src/db/states/state.rs [hunk 3]
```diff
@@ -34,7 +35,7 @@ pub struct State<DB: Database> {
     /// return not existing account and storage.
     ///
     /// Note: It is marked as Send so database can be shared between threads.
-    pub database: DB, //Box<dyn Database<Error = DBError> + Send + 'a>,
+    pub database: DB,
```
Comment (rakita) [inline crates/revm/src/db/states/state.rs:38]: ty

**Comment by rakita** [comment]
@DaniPopes if you are okay, would take this over

**Review by rakita — APPROVED** [review]
lgtm

**Pull request merged**